IDR/notebook/nbconverted/1.describe_screens.py

```diff
@@ -0,0 +1,284 @@
+
+    details_ = (
+        pd.DataFrame({x[0]: x[1]
+                      for x in annotations[study_index]["values"]}, index=[0])
+        .assign(
+            internal_id=id_,
+            upload_date=date_,
+            idr_name=name_,
+            screen_id=screen_id
+        )
+    )
+
+    return details_
+
+
+def describe_screen(screen_id):
+    """Pull additional metadata info per plate, given screen id
+
+    Parameters
+    ----------
+    screen_id: int
+        ID of the screen data set
+
+    Returns
+    -------
+    pandas.DataFrame() of the following metadata values:
+
+        screen_id: IDR ID for the screen
+        plate_id: IDR ID for each plate
+        plate_name: Names given to each plate
+        image_id: IDR ID for each image
+        cell_line: Cell line used in the screen experiment
+        strain: Strain of the cell line (if specified)
+        gene_identifier: Accession code for the gene being perturbed in a well
+        phenotype_identifier: Accession code for the phenotype perturbed in a well
+        stain: Set of the stains used in the screen
+        stain_target: The target protein or media of the stain
+        pixel_size_x: Width of the image
+        pixel_size_y: Height of the image
+        imaging_method: Method used to collect images (ex: fluorescence microscopy)
+    """
+    session = requests.Session()
+
+    # Get number of plates per screen and append to a dictionary
+    PLATES_URL = f"https://idr.openmicroscopy.org/webclient/api/plates/?id={screen_id}"
+    all_plates = session.get(PLATES_URL).json()["plates"]
+    study_plates = {x["id"]: x["name"] for x in all_plates}
+    print(f"Number of plates found in screen {screen_id}: ", len(study_plates))
+
+    # Initialize results list
+    plate_results = list()
+
+    # Iterate through all plates in the study
+    for plate in study_plates:
+        imageIDs = list()
+        plate_name = study_plates[plate]
+
+        # Access .json file for the plate ID. Contains image ID (id) numbers
+        # for replicate images per plate.
+        WELLS_IMAGES_URL = f"https://idr.openmicroscopy.org/webgateway/plate/{plate}/"
+        grid = session.get(WELLS_IMAGES_URL).json()
+
+        try:
+            pixel_size_x = grid["image_sizes"][0]["x"]
+            pixel_size_y = grid["image_sizes"][0]["y"]
+
+            # Get image and well ids
+            for image in grid['grid'][0]:
+                # Append IDs to iterable lists
+                thumb_url = image['thumb_url'].rstrip(image['thumb_url'][-1])
+                image_id = thumb_url.split('/')[-1]
+                imageIDs.append(image_id)
+
+        except (ValueError, KeyError):
+            plate_results.append(
+                [screen_id, plate, plate_name, None, None, None, None, None])
+            continue
+
+        # TODO: Rewrite to optimize and make consice
+        # TODO: Include cell/tissue label and map IDR accession to each screen
```

Comment:
TODO notes like these all too often get lost in the shuffle.

I've found that a more sustainable practice for these kinds of notes are to open a github issue (https://github.com/WayScience/publicly-available-microscopy-data/issues).

We also think about merging into the `main` (aka `master`) branch as code in complete state (i.e. a working state free of debugging print statements, TODO comments, etc.)

A complete state does not mean a final state, implying that the notebooks can never be improved. We hope to always improve!

Reply:
The TODO statements were removed from the production files and issues were added for each. See issues #3, #4,  #5, and #6. 

Added in 05a48d6d0afffb0a5009904c05024b7f2ca3aac6

My lab desktop (with VScode) has been updating for a few hours now and I'm unable to change the `.ipynb` files in my laptop PyCharm IDE. I will remove the TODO statements from the notebook files once I can access the desktop.

Reply:
The TODO statements in the `.ipynb` files were removed.

Added in 0367d5765a7aacc9861aa581ed4f8f0e1775ee95

Reply:
this is PERFECT! 💯